ppb_vector/vector2.py

```diff
@@ -288,6 +289,9 @@ def scale_to(self: VectorOrSub, length: Realish) -> VectorOrSub:
         if length < 0:
             raise ValueError("Vector2.scale_to takes non-negative lengths.")
 
+        if length == 0:
+            return type(self)(0, 0)
```

Comment:
I just noticed that the testsuite doesn't fail if we return `Vector2(0, 0)` there.

That's probably an opportunity to make more tests in `test_typing` use Hypothesis, so we can hit all codepaths in the methods under test, not the single one that's exercised by the hardcoded inputs, but that feels out-of-scope here, and like something that should be discussed (esp. since I have very little understanding of Python's object model).

@astronouth7303 @pathunstrom Please open a new issue for this if you think it's worth fixing.

Reply:
I'm ok with leaving the type-oriented stuff largely manual for now, since there's a relatively small number of possible types accepted, unless you think there's significant value in doing a hypothesis thing with it.

Reply:
See proposal in #101 

Given that moving those tests to Hypothesis didn't bring in extra complexity, and that it makes them much more useful (since we are now testing all code paths, instead of a fixed one), I think it would be reasonable to have.